pkg/apis/serving/v1alpha1/local_model_node_group_types.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2023 The KServe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LocalModelNodeGroupSpec defines a group of nodes for to download the model to.
+// +k8s:openapi-gen=true
+type LocalModelNodeGroupSpec struct {
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="persistentVolume is immutable"
+	PersistentVolumeSpec corev1.PersistentVolumeSpec `json:"persistentVolumeSpec"`
```

Comment:
What happens if user makes a mistake and the node group is already in use. Is it safe to delete and recreate?

Reply:
good question! I haven't thought about it before. Maybe we can ignore "node group not found" errors in the model cache reconciler. It would not affect existing models or services since the pvs and pvcs are per-model. Existing pvs and pvcs are not updated. However it would mess up with node storage space calculation, which we do not have yet.